api/src/routers/post.router.ts

```diff
@@ -0,0 +1,28 @@
+import express from 'express';
+import {
+  createPost,
+  createPostComment,
+  createPostTag,
+  getAllPosts,
+  getPostById,
+  getPostsByUser,
+  removePostLike,
+  removePostTag,
+  setPostLike,
+} from '../services/post.service';
+
+const postRouter = express.Router();
+
+postRouter.get('/', getAllPosts);
+postRouter.get('/:postId', getPostById);
+postRouter.get('/user/:userId', getPostsByUser);
+
+postRouter.post('/', createPost);
+postRouter.post('/like', setPostLike);
```

Comment:
These are duplicating business logic for post and comment. Better to separate it into standalone POST /likes route with sourceId to post/comment if you are threating like as separate entity with separate collection.